### ISSUE TITLE
About the test results

### ISSUE THREAD

**Issue by sunshine7362** [PuFanqi23/MonoDGP#10]
Hello, thank you for the open-source code. The results uploaded on the official website are the test results for 7518 samples from the test set, but when I reproduce the results using the code, it gives results for the 7480 samples from the training set. Where do I need to modify the configuration to obtain the test results from the test set?

**Comment by PuFanqi23** [comment]
Hi, Thanks for your interest in our work.
You need to set test_split: 'test' in monodgp.yaml and run eval code to generate test results. Remeber to set train_split: 'trainval' in monodgp.yaml if you want to train from scratch.

**Comment by sunshine7362** [comment]
Hello, thank you for your previous answers to my questions. I train using trainval and upload the test to the official website to get results. I noticed that the code saves the checkpoint from the last round. How can I ensure that the last checkpoint is the best one? This will also affect the testing results.

**Comment by PuFanqi23** [comment]
Since we can't evaluate on the test set during training, the standard practice is:

1. Save the checkpoint with the best validation performance.

2. Run multiple training sessions (with different seeds if needed) and submit each best validation model to the test server.

3. Report the best test result among these submissions.
